public/js/length_visualisation.js

```diff
@@ -0,0 +1,339 @@
+import _ from 'underscore';
+import VisualisationHelpers from './visualisation_helpers';
+
+export default class LengthVisualisation {
+  constructor(query, svg_container, algorithm) {
+    this._svg_container_d3 =  d3.select(svg_container[0]);
+    this._svg = {};
+    this._query = query;
+    this._seq_type = this.get_seq_type(algorithm);
+    this._hits = query.hits;
+    this.visualisation_helpers = new VisualisationHelpers();
+
+    this._margin = {top: 30, right: 60, bottom: 70, left: 25};
+    this._width = 838 - this._margin.left - this._margin.right;
+    this._height = 400 - this._margin.top - this._margin.bottom;
+
+    this._data = this.fill_data();
+    this._x_scale = d3.scale.linear()
+        .domain([
+          0,
+          (d3.max([this._query.length,d3.max(this._data)]) * 1.05)
+        ])
+        .range([0, this._width]);
+    this._bins = this.create_bins();
+    this._data2 = this.fill_data2();
+    this._y_scale = d3.scale.linear()
+        .domain([0, d3.max(this._bins, function(d) { return d.length })])
+        .range([this._height, 0]);
+    // console.log("Bin details (length) "+this._bins.length);
+    // console.log('new data2 trails (length) '+this._data2.length);
+    // this._data2.map(function (d) {
+    //   console.log('data2-1 length '+d.length);
+    //   console.log('its content '+d.toString());
+    // })
+    this._axis = this.create_axis();
+    this._svg.d3 = this._svg_container_d3.insert('svg', ':first-child')
+        .attr('width', this._width + this._margin.right + this._margin.left)
+        .attr('height', this._height + this._margin.top + this._margin.bottom)
+        .append('g')
+        .attr('transform','translate('+this._margin.left+','+this._margin.top+')');
+    this.setupTooltip();
+    this.draw_visualisation();
+  }
+
+  fill_data() {
+    var data = [];
```

Comment:
`map` returns an array. You are discarding `map`'s return value and using it like `each`.

`return _.map(this.query, _.iteratee('length'))`